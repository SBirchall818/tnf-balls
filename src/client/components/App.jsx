import React from 'react';
import '../normalize.css';
import '../app.css';
import {
  BrowserRouter as Router,
  Route,
  Switch,
  Redirect,
} from 'react-router-dom';
import { UsersScreen, BallsScreen, ConfigScreen } from '../containers';


const App = () => (
  <Router>
    <Switch>
      <Route
        exact
        path="/"
        component={UsersScreen}
      />
      <Route
        path="/config"
        component={ConfigScreen}
      />
      <Route
        path="/balls"
        component={BallsScreen}
      />
      <Route render={() => <Redirect to="/" />} />
    </Switch>
  </Router>
);


export default App;
