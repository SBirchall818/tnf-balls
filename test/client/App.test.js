import React from 'react';
import renderer from 'react-test-renderer';
// import { StyleSheetTestUtils } from 'aphrodite';
import { shallow } from 'enzyme';

import App from '../../src/client/App';

describe('App :: ', () => {
  // beforeEach(() => {
  //   StyleSheetTestUtils.suppressStyleInjection();
  // });

  // afterEach(() => {
  //   StyleSheetTestUtils.clearBufferAndResumeStyleInjection();
  // });

  it('should exist', () => {
    expect(App).toBeDefined();
  });

  it('should render App with defaults', () => {
    const component = renderer.create(<App />);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should use enzyme to shallow render', () => {
    const appComponent = shallow(<App />);
    const foundDiv = appComponent.find('div');
    expect(foundDiv.children()).toHaveLength(1);
  });
});