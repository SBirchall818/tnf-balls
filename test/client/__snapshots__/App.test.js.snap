// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App ::  should render App with defaults 1`] = `
<div>
  <h1>
    Loading.. please wait!
  </h1>
</div>
`;
